doc/kubernetes/modules/ROOT/pages/running/metrics/jgrp_metrics.adoc

```diff
@@ -0,0 +1,125 @@
+= Network Metrics
+:description: This documentation covers the information about the key network metrics providing by JGroups.
+
+{description}
+
+NOTE: {ispn} creates 2 clusters, one for intra-site communication and other for cross-site communication.
+The tag `cluster` identifies the cluster that the metric belongs.
+
+== Thread Pool
+
+When a message is received from another node, it is sent to a thread pool to be delivered to {ispn}.
+If no threads are available, the message is dropped, leading to retransmission and increase network usage.
+
+The metrics below report the thread pool size.
+
+TIP: In a healthy cluster, the thread pool should never be closer to its maximum size (by default, `200` threads).
+
+NOTE: Thread pool metrics are not available with virtual threads.
+
+|===
+|Metric |Description
+
+|`vendor_jgroups_tcp_get_thread_pool_size`
+|Current number of threads in the thread pool, for the intra-cluster communication.
+
+|`vendor_jgroups_tunnel_get_thread_pool_size`
+|Current number of threads in the thread pool, for the cross-site communication.
+
+|===
+
+== Flow Control
+
+Flow control takes care of adjusting the rate of a message sender to the rate of the slowest receiver over time.
+This is implemented through a credit based system, where each sender credits and decrements them whenever a message is sent.
+The sender blocks when the credits fall below 0, and only resumes sending messages when it receives a replenishment message from the receivers.
+
+The metrics below show the number of blocked messages and the average blocking time.
+When a value different from zero, it may signal that a receiver is overloaded and may degrade the cluster performance.
+
+Each node has two independent flow control protocols, `UFC` for unicast messages and `MFC` for multicast messages.
+Unicast messges are one-to-one messages while multicast are one-to-many messages.
+{ispn} mostly uses unicast messages.
```

Comment:
`{jdgserver_name}`

Reply:
Does the Infinispan operator allow to change the stack? In any case, the default is TCP.

Reply:
Yes it's TCP for the server, but I thought the default for Embedded was UDP still?

https://github.com/infinispan/infinispan/blob/cea9ef6ac5b50c9a8d5f3f95a3e9a9a432f5ffcf/core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java#L165

If we're targeting this for Keycloak then I think we need to explain both Embedded and Server.

> Does the Infinispan operator allow to change the stack

Only if you provide a custom configuration via a ConfigMap

Reply:
Well, when the issue was created, I thought it was for the external Infinispan cluster only. Because there is no embedded Infinispan in Keycloak; it does not start JGroups.